### PR TITLE
Adding --build-env to oc new-app documentation

### PR DESCRIPTION
**Pull request by coreydaley**
Adding description and example usage of the
oc new-app --build-env command

### CODE REVIEW

**Comment by coreydaley** [comment]
@bmcelvee ptal
@jim-minter @bparees fyi

File: dev_guide/application_lifecycle/new_app.adoc [hunk 1]
```diff
@@ -389,6 +389,41 @@ Any `BuildConfig` objects created as part of `new-app` processing will not be up
...
+
+----
+$ cat ruby.env | oc new-app openshift/ruby-23-centos7 --env-file=-
+----
+
+[NOTE]
+====
+Any `BuildConfig` objects created as part of `new-app` processing will not be updated with
+environment variables passed via the `-e|--env` or `--env-file` argument.
+====
```
Comment (bparees) [inline dev_guide/application_lifecycle/new_app.adoc]: probably don't need to repeat this note, it's covered in the --env section above.

Comment (coreydaley) [inline dev_guide/application_lifecycle/new_app.adoc]: Updated

File: dev_guide/application_lifecycle/new_app.adoc [hunk 2]
```diff
@@ -389,6 +389,35 @@ Any `BuildConfig` objects created as part of `new-app` processing will not be up
...
+HTTP_PROXY=http://myproxy.net:1337/
+GEM_HOME=~/.gem
+$ oc new-app openshift/ruby-23-centos7 --build-env-file=ruby.env
+----
+
+Additionally, environment variables can be given on standard input by using
+`--build-env-file=-`:
+
+----
+$ cat ruby.env | oc new-app openshift/ruby-23-centos7 --env-file=-
```
Comment (bparees) [inline dev_guide/application_lifecycle/new_app.adoc]: should be --build-env-file

Comment (coreydaley) [inline dev_guide/application_lifecycle/new_app.adoc]: Doh!  I was sure I had changed them all.  Updated

**Comment by bparees** [comment]
lgtm.  @coreydaley do you remember if this was part of 3.4, or was 3.5 the first release?  please update the labels if needed.

**Comment by coreydaley** [comment]
@bparees I just checked origin 1.4 and --build-env was not there, so I think the labels are good.

**Comment by jim-minter** [comment]
lgtm

**Comment by coreydaley** [comment]
@bmcelvee Looks like we are good from our side, let me know if it needs any updates

**Comment by bmcelvee** [comment]
Thanks, @coreydaley!

**Comment by bmcelvee** [comment]
[rev_history]
|xref:../dev_guide/application_lifecycle/new_app.adoc#dev-guide-new-app[Creating New Applications]
|Added xref:../dev_guide/application_lifecycle/new_app.adoc#specifying-build-environment-variables[Specifying Build Environment Variables] section.
%

**Pull request merged**